### PR TITLE
Refactor and test `AddAttachment`

### PR DESCRIPTION
**Pull request by kavir1698**

### CODE REVIEW

File: datasetIngestor/addAttachment.go [hunk 1]
```diff
@@ -11,27 +11,28 @@ import (
...
+func ReadAndEncodeImage(attachmentFile string) (string, error) {
 	imgFile, err := os.Open(attachmentFile)
 	if err != nil {
-		log.Fatalf("Can not open attachment file %v \n", attachmentFile)
+		return "", err
 	}
-
 	defer imgFile.Close()
-
+
```
Comment (minottic) [inline datasetIngestor/addAttachment.go:20]: I wonder why the linting doesn't complain here

**Review by minottic — APPROVED** [review]
very minor comment. Looks good!

File: datasetIngestor/addAttachment.go (hunk 1 above)
Comment (minottic) [reply datasetIngestor/addAttachment.go:20]: I thought this wasn't just an empty line, but an empty line with an extra `tab`

Comment (kavir1698) [reply datasetIngestor/addAttachment.go:20]: it is with a tab. But all my code is like that. That's how vscode does it by default

Comment (minottic) [reply datasetIngestor/addAttachment.go:20]: ok, that's weird, I thought lint would complain about that. Well not a problem anyway...

**Pull request merged**